colibris/views.py

```diff
@@ -1,17 +1,198 @@
+
+        result = schema.dump(item)
+
+        return web.json_response(result, status=201)
+
+
+class RetrieveMixin(metaclass=_GenericMixinMeta):
+    model = Model
+    schema_class = ModelSchema
+
+    async def get(self):
+        schema = self.get_schema()
+        instance = self.get_object()
+
+        result = schema.dump(instance)
+
+        return web.json_response(result)
+
+
+class UpdateMixin(metaclass=_GenericMixinMeta):
+    model = Model
+    schema_class = ModelSchema
+
+    async def patch(self):
+        schema = self.get_schema(partial=True)
+        data = await self.get_validated_data(schema)
+
+        instance = self.get_object()
+        instance.update_fields(data)
+
+        try:
+            instance.save(only=data.keys())
+        except IntegrityError as err:
+            traceback.print_exc()
+            raise api.ServerError(code='server_error', message=str(err))
+
+        result = schema.dump(instance)
+
+        return web.json_response(result)
+
+    async def put(self):
+        schema = self.get_schema()
+        data = await self.get_validated_data(schema)
+
+        instance = self.get_object()
+        instance.update_fields(data)
+
+        try:
+            instance.save()
+        except IntegrityError as err:
+            traceback.print_exc()
+            raise api.ServerError(code='server_error', message=str(err))
+
+        result = schema.dump(instance)
+
+        return web.json_response(result)
+
+
+class DestroyMixin:
+    async def delete(self):
+        instance = self.get_object()
+        instance.delete_instance()
+
+        return web.json_response(status=204)
+
+
+class BaseModelView(web.View):
+    model = Model
+    schema_class = ModelSchema
+    url_identifier = 'id'
+    lookup_field = 'id'
+
+    def get_query(self):
+        return self.model.select().order_by(self.model.id.desc())
+
+    def get_object(self):
+        identifier_value = self.request.match_info[self.url_identifier]
+
+        try:
+            instance = self.get_query().where(getattr(self.model, self.lookup_field) == identifier_value).get()
+        except self.model.DoesNotExist:
+            raise api.ModelNotFoundException(self.model)
+
+        return instance
+
+    def get_schema(self, *args, **kwargs):
+        kwargs.update({
+            'context': {'request': self.request}
+        })
+
+        schema = self.schema_class(*args, **kwargs)
+
+        return schema
+
+    async def get_validated_data(self, schema):
+        json_payload = await self.get_request_payload()
+
+        try:
+            data = schema.load(json_payload)
+        except ValidationError as err:
+            raise api.InvalidRequest(code='invalid_request', message=err.messages)
+
+        return data
+
+    async def get_request_payload(self):
+        try:
+            json_payload = await self.request.json()
+        except JSONDecodeError:
+            raise api.JSONParseError()
```

Comment:
@ccrisan, is this better?

Reply:
Amost. `JSONParseError` should inherit `InvalidRequest`.

Reply:
Ok, agreed.